File: 57-insert-interval/insert-interval.cpp
class Solution {
public:
    vector<vector<int>> insert(vector<vector<int>>& vals, vector<int>& nVal) {
        int n=(int)vals.size();
        if(n==0){
            vals.push_back(nVal);
            return vals;
        }
        else if(vals[0][0]>nVal[1]){
            vals.insert(vals.begin()+0, nVal);
            return vals;
        }
        else if(vals[n-1][1]<nVal[0]){
            vals.push_back(nVal);
            return vals;
        }
        for(int i=0;i<n;++i){
            if(nVal[0]<=vals[i][1]){
                vals.insert(vals.begin()+i, nVal);
                break;
            }
        }
        for(int i=1;i<=n;++i){
            if(vals[i][0]<=vals[i-1][1]){
                vals[i-1][0]=min(vals[i-1][0], vals[i][0]);
                vals[i-1][1]=max(vals[i-1][1], vals[i][1]);
                vals.erase(vals.begin()+i);
                --i;
                --n;
            }
            
        }
        return vals;
    }
};

File: 57-insert-interval/insert-interval.java
class Solution {
    public int[][] insert(int[][] intervals, int[] newInterval) {
        int i=0, n= intervals.length, siz;
        List<int[]> list = new ArrayList<>();
        while(i<n && intervals[i][1]<newInterval[0]){
            list.add(intervals[i++]);
        }
        while(i<n && intervals[i][0]<=newInterval[1]){
            newInterval[0]= Math.min(newInterval[0], intervals[i][0]);
            newInterval[1]= Math.max(newInterval[1], intervals[i][1]);
            i++;
        }
        list.add(newInterval);
        while(i<n){
            list.add(intervals[i++]);
        }
        siz= list.size(); i=0;
        int[][] ans= new int[siz][2];
        for(i=0;i<siz;i++) ans[i]= list.get(i);
        return ans;
    }
}